rust/src/main.rs

```diff
@@ -80,14 +86,15 @@ fn gen(args: Args) -> Result<()> {
 
     // Read protobuf header
     let header = input.read_message::<Header>()?;
-    let num_documents = header.get_num_docs();
+    let num_documents =
+        u32::try_from(header.get_num_docs()).expect("Number of documents must be non-negative.");
```

Comment:
@JMMackenzie I changed a few lines and I thought you might be interesting why. This is the first of them, I use `TryFrom` trait's `try_from` function here, which `u32` implements on `i32`. Essentially, it will return `Err` if can't construct, so when it's negative. Just a little bit of code to explicitly fail instead of a generic panic. In most situations you'd propagate this error with `?` but it doesn't really matter to us here (we'll fail anyway), and this gives me a way to put a custom error message. Ideally, you'd use one of the error handling libraries, and call something like `.context("...")?` instead of expect but hey, I'm lazy :D